### DISCUSSION TITLE
x-mask money plugin does not allow for negative values

### DISCUSSION THREAD

**Discussion by benkoshy** [alpinejs/alpine#3324] [3. Feature ideas]
Sometimes it's useful to model amounts in negative values. 

This is especially useful in modelling losses (as opposed to profits):

`-10,000`.

Currently this cannot be done: https://alpinejs.dev/plugins/mask#money-inputs - try here inputting a negative value.

I tried cloning the repository and running tests to make a PR for this, but the test seem to fail against the main branch? 

https://github.com/alpinejs/alpine/tree/5ede2501b70533f067796992c8d379d2ddd07af9

![image](https://user-images.githubusercontent.com/15097447/206370217-87420154-1c6d-4b42-bb4d-bd70fbc8112f.png)

**Comment by SimoTod** [comment]
The tests run in cypress. I'm not sure if the jest ones are still in use.

**Reply by benkoshy** [reply]
chrs, the jest ones are the ones that are failing.

I am proposing that we allow for negative numbers on the money mask - is that something that is unexpected behaviour? Would maintainers be opposed to such a PR?

**Reply by ekwoka** [reply]
I think Caleb would be open to it

**Reply by medspec** [reply]
is there any movement on this?

**Reply by benkoshy** [reply]
Unfortunately not yet

**Reply by JanderV** [reply]
Here's a pull request https://github.com/alpinejs/alpine/pull/3416

**Reply by benkoshy** [reply]
well done!  (was on my list of things to-do, i could never get back to finishing it).

Looks like the build is failing, but if you see it through to completion, you would have solved some headaches for a lot of people.

**Reply by JanderV** [reply]
Thanks! I don't think I can do much about that failing `plugins/ui/popover.spec.js` test. 
From what I've read it's known to be quite flaky (https://github.com/alpinejs/alpine/discussions/3394, https://github.com/alpinejs/alpine/pull/3312, https://github.com/alpinejs/alpine/pull/3339#issuecomment-1360465213), so hopefully the PR still gets merged.
